docs-src/python/how-to-use-a-custom-payload-codec-in-python.md

```diff
@@ -0,0 +1,72 @@
+---
+id: how-to-use-a-custom-payload-codec-in-python
+title: How to use a custom Payload Codec in Python
+sidebar_label: Custom Payload Codec
+description: Create a custom `PayloadCodec` implementation and define your encryption/compression and decryption/decompression logic in the `encode` and `decode` functions.
+tags:
+  - python
+  - developer-guide
+---
+
+**Create a custom Payload Codec**
+
+Custom Data Converters can change the default Temporal Data Conversion behavior by adding hooks, sending payloads to external storage, or performing different encoding steps.
+If you only need to change the encoding performed on your payloads -- by adding compression or encryption -- you can override the default Data Converter using by creating a new `PayloadCodec`.
+
+The `PayloadCodec` needs to implement `encode()` and `decode()` functions at a minimum.
+These should loop through all of a Workflow's payloads, perform all of your necessary marshalling, compression, or encryption steps in order, and set an `"encoding"` metadata field.
+Here is an example that marshals and then compresses a payload using Python's [cramjam](https://github.com/milesgranger/cramjam/tree/master/cramjam-python) library to provide `snappy` compression:
+
+```python
```

Comment:
shouldn't this code be external to docs and imported in so we can test it?

Reply:
Yes... but I don't think we're using snipsync consistently enough in this repo right now, and most of these snippets are fairly abstracted and I don't think we could make them pass any CI from top to bottom.